source/credentials_provider_sts.c

```diff
@@ -334,16 +336,21 @@ static void s_on_connection_setup_fn(struct aws_http_connection *connection, int
         .on_complete = s_on_stream_complete_fn,
     };
 
-    struct aws_http_stream *stream =
-        provider_impl->function_table->aws_http_connection_make_request(connection, &options);
-    if (!stream) {
-        goto error;
-    }
+    stream = provider_impl->function_table->aws_http_connection_make_request(connection, &options);
+    if (stream) {
+        if (provider_impl->function_table->aws_http_stream_activate(stream)) {
+            goto error;
+        }
 
-    provider_impl->function_table->aws_http_stream_release(stream);
+        provider_impl->function_table->aws_http_stream_release(stream);
+        return;
+    }
 
-    return;
 error:
+    if (stream) {
+        provider_impl->function_table->aws_http_stream_release(stream);
+    }
```

Comment:
it's safe to call release() on a NULL stream, to support naive cleanup in `error:` labels like this